[Kevin Foong] iP

kevin9foong opened the pull request:
# Duke 🦑 (header)
### Kevin's CS2103T iP 👍

Features (bullet list): 
* Add Task Variations (Event, Deadline, ToDo)  
* Delete Tasks
* Search for Tasks 
* ~~GUI~~ CLI-based 💪 

Product Roadmap 🚀 (numbered list): 
1. Create core features
2. Learn new stuff 
3. Draw UML diagrams 💎 
4. **Have fun!** 

Fenced code block: 
```
system.out.println("Hi, I'm Duke at your service");
```

Tasks remaining: 
- [x] Week 4
- [x] Week 5
- [x] Week 6
- [x] _and beyond..._ 

As _Edgar W. Dijkstra_ once said ☄️ :
> Perfecting oneself is as much unlearning as it is learning. 

Hyperlink: 
[HttpStatusDogs 🐶 ](https://httpstatusdogs.com/)

This is some inline code `system.out.println("Hi, I am an inline code.");`.

Commit:
Add Gradle support

kevin9foong committed:
Add skeletal functionality with Bye and Echo commands

kevin9foong committed:
Add list and add functionality

kevin9foong committed:
Add ByeMessage and GreetingMessage classes

kevin9foong committed:
Add mark as done functionality

kevin9foong committed:
Add Todo, Event and Deadline support

kevin9foong committed:
Add testing using I/O redirection technique

kevin9foong committed:
Add error handling and update test cases

kevin9foong committed:
Add delete functionality and test cases

kevin9foong committed:
Add enums to represent Commands available

kevin9foong committed:
Add Save functionality by persisting data in txt file

kevin9foong committed:
Add Date support for Deadlines

kevin9foong committed:
Update IO redirection test cases

kevin9foong committed:
Merge branch 'branch-Level-7'

kevin9foong committed:
Add Date support

kevin9foong committed:
Refactor code to meet OOP requirements

kevin9foong committed:
Organize classes into packages

kevin9foong committed:
Add Junit tests for TaskList and AddDeadlineCommand

kevin9foong committed:
Make test classes default access mod based on recommendation

kevin9foong committed:
Remove duke/ from the Task delete and add messages

kevin9foong committed:
Update .gitignore to exclude META-INF directory

kevin9foong committed:
Add javadocs for public methods

kevin9foong committed:
Follow Java style guide: rename associated constants, organise imports

kevin9foong committed:
Add find functionality

kevin9foong committed:
Merge branch 'branch-A-JavaDoc'

kevin9foong committed:
Merge branch 'branch-A-CodingStandard'

kevin9foong committed:
Merge branch 'branch-Level-9'

kevin9foong committed:
Update javadocs for increased fluency

kevin9foong committed:
Add JavaDocs header comments

kevin9foong committed:
Merge branch 'add-gradle-support' of https://github.com/kevin9foong/ip into week-4-parallel

kevin9foong committed:
Add exception handling for invalid task save file data

kevin9foong committed:
Update delimiter for file representation to something less common

kevin9foong committed:
Add Checkstyle config files and reformat code to follow code style

kevin9foong committed:
Add precommit task to gradle to clean, build, run tests and checkstyle

kevin9foong committed:
Add a simple GUI with JavaFX

kevin9foong committed:
Update resizing behaviour for TextArea in GUI

kevin9foong committed:
Merge branch 'branch-Level-10'

kevin9foong committed:
Add GUI for commands

kevin9foong committed:
Merge branch 'branch-Level-10'

kevin9foong committed:
Add javadocs

kevin9foong committed:
Merge branch 'branch-Level-10'

kevin9foong committed:
Edit behaviour of Duke to exit immediately upon receiving bye command

fans2619 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,203 @@
+        primaryStage.setScene(scene);
+        primaryStage.show();
+    }
+}
```
Do you want to insert a newline at the end of every java source file?

fans2619 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,84 @@
+import duke.exceptions.InvalidTaskDataException;
+
+/**
+ * This class encapsulates a user-added task
```
Probably missing "." at the end.

fans2619 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    /**
+     * Set description of this task.
```
Should it be "Sets"?

kevin9foong replied:
Resolved. Thanks for pointing it out!

fans2619 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    /**
+     * Convert this task's data to representation to be saved in file.
```
Should it be "Converts"?

fans2619 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    /**
+     * Get task associated with index number in taskList.
```
Should it be "Gets"?

fans2619 reviewed (commented):
Yea it is excellent in terms of coding standard. Good job! 👍

Nature711 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,39 @@
+     * @throws DukeException thrown when a failure occurs within the Duke chat bot.
+     */
+    public abstract void execute(UserOutputHandler userOutputHandler, TaskList taskList)
+            throws DukeException;
```
I feel like there's no need to put the throws part in a new line; as I've counted, putting the entire method signature in one line won't make the line longer than 120 characters :D

Nature711 reviewed (commented):
LGTM. Just need to fix some minor issues in the tense used in javadoc :D

Moley456 reviewed (commented):
Everything looks good! Clean and good coding in general, just a small issue with the code related to the GUI and very minor issues with JavaDocs comments that have been brought up by the other reviewers. Good job and keep it up! 👍🏻

kevin9foong committed:
Fix javadoc styling according to comments

kevin9foong committed:
Update GUI to follow the JavaFX guide

kevin9foong committed:
Add use of varargs

kevin9foong committed:
Add assertions to verify assumptions

The codebase does not include any assertions.

Adding assumptions allows us to improve code robustness by
detected potential programmer mistakes during runtime.

Add assertions at appropriate places, avoid asserting
public method arguments according to Sun best practices.

Assertions can be disabled and hence checking
public arguments with assertions can cause
errors to go undetected.

kevin9foong committed:
Improve coding quality and refactor code

There are places where code quality can be improved
such as the existence of unused code.

Needs to change to enhance readability and prevent potential bugs.

Let's follow coding readability and refactoring guidelines:
* Give complicated expressions meaningful names
* Avoid magic literals
* Removing dead code

These guidelines are specifically used as they clean up code
clutter and make our code more readable.

kevin9foong committed:
Merge pull request #1 from kevin9foong/branch-A-Assertions

Add assertions to verify assumptions

kevin9foong committed:
Rename generateMainWindow method

kevin9foong committed:
Switch to FXML

Not using FXML, leading to mixing UI and Chat bot logic.
Also, start() method exceeds 30 LOC per method code quality
guideline.

Let's change to using JavaFX FXML and remove newly unused
functionality.

Using FXML allows us to separate UI and Chat bot logic, leading
to more maintainable code.

Also, removed use of varargs since it is currently no longer
needed.

kevin9foong committed:
Merge branch 'master' into branch-A-CodeQuality

kevin9foong committed:
Merge pull request #2 from kevin9foong/branch-A-CodeQuality

Improve coding quality and refactor code

kevin9foong committed:
Add usage of streams

kevin9foong committed:
Merge pull request #3 from kevin9foong/branch-A-Streams

Add usage of streams

kevin9foong committed:
Add CI Gradle check

kevin9foong committed:
Improve implementation of loadCustomAvatarImagesFromArgs method

kevin9foong committed:
Add reminders feature

kevin9foong committed:
Merge pull request #4 from kevin9foong/branch-B-Reminders

Add reminders feature

kevin9foong committed:
Refactor ReminderCommand execute method

kevin9foong committed:
Improve GUI, fix delete functionality, update reminder message

kevin9foong committed:
Add user guide and product screenshot

kevin9foong committed:
Include space in list and find message and update font to support macOS

kevin9foong committed:
Fix javadocs

kevin9foong committed:
Fix shutdown behaviour for corrupted save file

kevin9foong committed:
Fix javadoc comment for main method

kevin9foong committed:
Remove public access for test and add final to image avatar fields
